api/v1beta1/metal3machine_types.go

```diff
@@ -99,9 +99,11 @@ func (s *Metal3MachineSpec) IsValid() error {
 	if err != nil {
 		invalid = append(invalid, "Image.URL")
 	}
-	_, err = url.ParseRequestURI(s.Image.Checksum)
-	if err != nil {
-		invalid = append(invalid, "Image.Checksum")
+	if s.Image.DiskFormat == nil || *s.Image.DiskFormat != "live-iso" {
```

Comment:
I am not quite sure about this, if the DiskFormat is nil, do we even have to validate the image.Checksum?

Reply:
This is a bad code, I changed it, PTAL.

Reply:
I checked the document again, and the value of `DiskFormat` can be unset, and the value of `DiskFormat` is not set in the corresponding UT. The validation of nil is added here to avoid null pointer errors in *s.Image.DiskFormat.